### PR TITLE
Added code changes for SNAP-3120

### PR DESCRIPTION
**Pull request by smahajan05**
## What changes were proposed in this pull request?
Added code changes to remove uri from ConcurrentHashMap variable 'files' while undeploying package.

## How was this patch tested?

Added dunit tests

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/rpc/netty/NettyStreamManager.scala [hunk 1]
```diff
@@ -65,6 +65,13 @@ private[netty] class NettyStreamManager(rpcEnv: NettyRpcEnv)
     }
   }
 
+  override def removeFile(file: File): Unit = {
+    val existingPath = files.putIfAbsent(file.getName, file)
```
Comment (ashetkar) [inline core/src/main/scala/org/apache/spark/rpc/netty/NettyStreamManager.scala]: Why not directly remove the file?

**Review by ashetkar — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/rpc/netty/NettyStreamManager.scala [hunk 2]
```diff
@@ -65,6 +65,10 @@ private[netty] class NettyStreamManager(rpcEnv: NettyRpcEnv)
     }
   }
 
+  override def removeFile(file: File): Unit = {
+    files.remove(file.getName)
```
Comment (kneeraj) [inline core/src/main/scala/org/apache/spark/rpc/netty/NettyStreamManager.scala]: It looks like making 'File' as argument is redundant. File path is all which is needed. Please check and change the method signatures accordingly.

**Review by kneeraj — CHANGES_REQUESTED** [review]
Please look at the comment and make the corresponding changes in other repositories as well.

File: core/src/main/scala/org/apache/spark/rpc/netty/NettyStreamManager.scala (hunk 2 above)
Comment (smahajan05) [reply core/src/main/scala/org/apache/spark/rpc/netty/NettyStreamManager.scala]: Here I need file name only not file path. File path is providing absolute path if i am keeping it as string. For retrieving only the file name from absolute path i am making File from string path.

Comment (kneeraj) [reply core/src/main/scala/org/apache/spark/rpc/netty/NettyStreamManager.scala]: @smahajan05 I understand that. What I am saying is that you don't need to create File object as such.
You are first creating path and then creating file from that and then just using the name of the file.
WIll this work?

Paths.get("whatever_str").getFileName()

I suggested this minor change only.

**Review by kneeraj — APPROVED** [review]
I think it is ok if remove returns 'null'
Just check if you need to take care of the return value as well or is it ok to ignore.

**Pull request merged**